shielder/cli/src/contract.rs

```diff
@@ -77,29 +77,28 @@ impl Shielder {
         new_note: FrontendNote,
         proof: &[u8],
     ) -> Result<u32> {
-        let new_note_bytes = bytes_from_note(&new_note);
-        let merkle_root_bytes = bytes_from_note(&merkle_root);
-        let old_nullifier_bytes = bytes_from_note(&old_nullifier);
-
-        let args = [
-            &*token_id.to_string(),
-            &*value.to_string(),
-            &*recipient.to_string(),
-            &*format!("{:?}", Some(fee_for_caller)),
-            &*format!("0x{}", hex::encode(merkle_root_bytes)),
-            &*format!("0x{}", hex::encode(old_nullifier_bytes)),
-            &*format!("0x{}", hex::encode(new_note_bytes)),
-            &*format!("0x{}", hex::encode(proof)),
-        ];
-
-        debug!("Calling withdraw tx with arguments {:?}", &args);
-        let tx_info = self
-            .contract
-            .contract_exec(connection, "withdraw", &args)
+        let ink_contract: Instance = (&self.contract).into();
+        let inner: [u8; 32] = *recipient.as_ref();
+        let ink_recipient: ink_primitives::AccountId = inner.into();
```

Comment:
yeah, I would move these account castings somewhere to a single place

Reply:
Same as above.